Zend/zend_constants.c

```diff
@@ -506,7 +506,7 @@ ZEND_API int zend_register_constant(zend_constant *c)
 		|| (!persistent && zend_get_special_const(ZSTR_VAL(name), ZSTR_LEN(name)))
 		|| zend_hash_add_constant(EG(zend_constants), name, c) == NULL
 	) {
-		zend_error(E_NOTICE,"Constant %s already defined", ZSTR_VAL(name));
```

Comment:
I don't think we can reclassify notices to exceptions :/

Reply:
We had a majority for changing a notice to an exception in the engine warnings RFC so I would argue there is precedent, and changing define in this way would be significantly less likely to cause disruption as redefining a constant has zero legitimate reason.

Reply:
Notices are also errors and are enabled in error_reporting in php.ini-production. So this is 👍 from me.